tools/packaging/packaging.go

```diff
@@ -103,48 +109,97 @@ func CreateLinuxPackages(osqueryVersion, hostname, secret string, insecure, inse
 		}
 	}
 
-	// Create the systemd unit file for the launcher service
-	systemdPath := filepath.Join(systemdDirectory, "launcher.service")
-	systemdFile, err := os.Create(filepath.Join(packageRoot, systemdPath))
-	if err != nil {
-		return "", "", errors.Wrap(err, "could not create launcher systemd unit file")
-	}
-	defer systemdFile.Close()
+	if systemd {
```

Comment:
Would you mind refactoring the changes into helper functions? Each branch here is so long that IMO it's hurting readability a bit.